query/src/servers/http/clickhouse_handler.rs

```diff
@@ -123,6 +124,14 @@ pub async fn clickhouse_handler_get(
         .await
         .map_err(InternalServerError)?;
 
+    if let Some(settings) = params.settings {
+        let session_settings: Vec<&str> = settings.split_terminator(',').collect();
+        for session_setting in session_settings.into_iter() {
+            let s: Vec<&str> = session_setting.split('=').collect();
+            session.get_settings().set_settings(s[0].to_string(), s[1].to_string(), false).map_err(BadRequest)?;
+        }
```

Comment:
some suggestion: 

1.  ` map_err(BadRequest)?;`   desc the wrong setting
2.  strip of key/value strings (remove spaces)

Reply:
Good idea. What do you mean `strip of key/value strings (remove spaces)?` We should better use ' ' or '-' as separator？

Reply:
e.g. allow `%20` after `, `

Reply:
Got it.

Reply:
Why not follow the param rule as ClickHouse ?

`http://localhost:8123/?max_result_bytes=4000000&buffer_size=3000000&wait_end_of_query=...?

Reply:
Why not follow the param rule as ClickHouse ?

`http://localhost:8123/?max_result_bytes=4000000&buffer_size=3000000&wait_end_of_query=...?

Reply:
This means that we put all settings in params? It seems not to maintain easily?

Reply:
https://github.com/ClickHouse/ClickHouse/blob/8b4a6a24161c701ff7255bc14f1e39b5f896049d/src/Server/HTTPHandler.cpp

See: `SettingsChanges settings_changes;`

Reply:
Yes, I need to check whether poem can receive any undefined key-value. Current method can't do this.

Reply:
https://serde.rs/attr-flatten.html#capture-additional-fields